src/Microsoft.Extensions.ObjectPool/DefaultObjectPool.cs

```diff
@@ -100,10 +95,6 @@ private void ReturnViaScan(T obj)
         private struct ObjectWrapper
```

Comment:
Can you add a comment here saying the `struct` is a performance optimization and shouldn't be simplified? It'd look very tempting given this that it's a struct with one field on it.

Reply:
Sure, this is one that people aren't really aware of.

Reply:
`// PERF: the struct wrapper avoids array-covariance-checks from the runtime`

Maybe something like this.